test/runner.js

```diff
@@ -79,6 +80,24 @@ module.exports = createRunner({
   reporter: process.env.MOCHA_REPORTER || 'list',
   overrideTestPaths: [/spec$/, /test/],
 }, mocha => {
+  // Ensure that we expect to be deployable to this version of Atom.
+  const engineRange = require('../package.json').engines.atom;
+  const atomEnv = global.buildAtomEnvironment();
+  const atomVersion = atomEnv.getVersion();
+  const atomReleaseChannel = atomEnv.getReleaseChannel();
+  atomEnv.destroy();
+
+  if (!semver.satisfies(semver.coerce(atomVersion), engineRange)) {
+    process.stderr.write(
+      `Atom version ${atomVersion} does not satisfy the range "${engineRange}" specified in package.json.\n` +
+      `This version of atom/github is currently incompatible with the ${atomReleaseChannel} ` +
+      'Atom release channel.\n',
+    );
+    // A regular expression that matches nothing.
```

Comment:
just for my own education: why do we need a regex that matches nothing?

Reply:
Well... I'm trying to find a graceful way to "cancel" the current test run and exit with a 0 status from the test runner. Mocha's `.grep()` method lets you filter the tests that execute by matching their titles against a regular expression, so providing it a regexp that will never match is equivalent to saying "don't run anything."

It's kind of awkward though. And I think it causes some misleading output from Mocha, although I tried to mitigate that with the `process.stderr` lines up above.

I tried doing a `process.exit(0)` first, but that caused the main process to think the renderer process had crashed and exit with a 1. Before that I was going to do this in a standalone script called from `script/cibuild`, but AppVeyor and CircleCI are both using the [atom/ci build scripts](https://github.com/atom/ci) which are hard-coded to run `atom --test test/`, so the test runner is the earliest common place to put the version check logic.

Now I'm thinking the Right Thing To Do :tm: is to do a little work upstream in atom-mocha-test-runner to support a clean early exit.